plan/task/exec.go

```diff
@@ -305,3 +308,20 @@ func (t *execTask) mountSocket(pctx *plancontext.Context, dest string, mnt *comp
 		llb.SSHSocketTarget(dest),
 	), nil
 }
+
+func (t *execTask) mountFile(_ *plancontext.Context, dest string, mnt *compiler.Value) (llb.RunOption, error) {
+	contents, err := mnt.Lookup("contents").String()
+	if err != nil {
+		return nil, err
+	}
+
+	permissions, err := mnt.Lookup("permissions").Int64()
+	if err != nil {
+		return nil, err
+	}
+
+	return llb.AddMount(
+		dest,
+		llb.Scratch().File(llb.Mkfile(dest, fs.FileMode(permissions), []byte(contents))), llb.SourcePath(dest),
```

Comment:
We should put the file created on scratch at a constant location, i.e. like this:
```
return llb.AddMount(
   dest,
   llb.Scratch().File(llb.Mkfile("/file", fs.FileMode(permissions), []byte(contents))),
   llb.SourcePath("/file"),
)
```

`/file` is arbitrarily chosen, any path under `/` works. This works the same way as using `dest` everywhere but has the advantage that if two mounts with the same file contents but different destinations are requested, Buildkit will only have to create the underlying file once and then just bind mount it to whatever path it's needed at.

Reply:
Agree, I was just thinking about that.

Reply:
Fixed in https://github.com/dagger/dagger/pull/2290/commits/37460d88d1ac176fecaa0433de5cd57897c6e3a0